src/main/java/edu/hm/hafner/analysis/parser/EclipseXMLParser.java

```diff
@@ -82,4 +106,92 @@ public Report parse(final ReaderFactory readerFactory) throws ParsingException {
         }
     }
 
+    /**
+     * These categories were taken from the ECJ source code. From March 25th, 2019, Ver. 3.18.
+     * 
+     * @param categoryId
+     *     eclipse generated category id.
+     * @return decoded category, or empty string.
+     * @see https://github.com/eclipse/eclipse.jdt.core/blob/master/org.eclipse.jdt.core/compiler/org/eclipse/jdt/core/compiler/CategorizedProblem.java
+     */
+    private String decodeCategory(String categoryId) {
```

Comment:
Nice - they already provide a category at least in the XML!

Reply:
They are handy, even if some of those category names are a mouthful.